src/components/Buttons/ButtonsStatusUpdate/ButtonsStatusUpdate.jsx

```diff
@@ -1,11 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
 import { Button } from '../Button/Button';
-import { BUTTONS_NAMES, BUTTONS_TYPES } from '../../../shared/constants';
-
-export function ButtonsStatusUpdate() {
-  return (
-    <div>
-      <Button title={BUTTONS_NAMES.success} stylingType={BUTTONS_TYPES.typeSave} />
-      <Button title={BUTTONS_NAMES.fail} stylingType={BUTTONS_TYPES.typeDelete} />
-    </div>
-  );
+import { BUTTONS_NAMES, BUTTONS_TYPES, STATUS_KEYS } from '../../../shared/constants';
+import { changeTaskStatus } from '../../../services/tasks-services';
+
+export class ButtonsStatusUpdate extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      status: '',
+    };
+  }
+
+  componentDidMount() {
+    const { status } = this.props;
+    this.setState({ status });
+  }
+
+  changeStatusHandler = async (newStatus) => {
+    const { taskId, userId, updateStateHandler } = this.props;
+    const updatedStatuses = await changeTaskStatus(taskId, userId, newStatus);
+    this.setState({ status: newStatus });
+    updateStateHandler(updatedStatuses, taskId);
+  };
+
+  render() {
+    const { status } = this.state;
+
+    return (
+      <>
+        {status === STATUS_KEYS.active ? (
+          <Button
+            title={BUTTONS_NAMES.success}
+            stylingType={BUTTONS_TYPES.typeSave}
+            onClick={() => {
+              this.changeStatusHandler(BUTTONS_NAMES.success);
```

Comment:
why do you need call it as `arrow function` here and below ?

Reply:
```suggestion
            onClick={this.succesStatusHandler}
```